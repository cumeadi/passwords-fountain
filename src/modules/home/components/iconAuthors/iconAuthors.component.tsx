import { h, VNode } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import {
    Wrapper,
    Heading,
    IconAuthorList,
    IconInfo,
} from './iconAuthors.styles';
import { Text } from '@/modules/localisation/localisation.context';

interface IconAuthor {
    id: number;
    iconName: string;
    author: string;
}

const iconAuthors: IconAuthor[] = [
    { id: 0, iconName: 'Padlock icon', author: 'Pixel perfect' },
    { id: 1, iconName: 'Success icon', author: 'Alfredo Hernandez' },
    { id: 2, iconName: 'Error icon', author: 'Freepik' },
    { id: 3, iconName: 'Fountain icon', author: 'Smashicons' },
    { id: 4, iconName: 'Database icon', author: 'Smashicons' },
    { id: 5, iconName: 'Home icon', author: 'Dave Gandy' },
    { id: 6, iconName: 'Heart icon', author: 'Smashicons' },
];

export const IconAuthors: TypedComponent<Props> = () => {
    const renderAuthors = (): VNode[] =>
        iconAuthors.map(({ id, iconName, author }) => (
            <IconInfo key={id}>
                {iconName} - {author}
            </IconInfo>
        ));

    return (
        <Wrapper>
            <Heading>
                <Text>home.thanksToIconAuthors</Text>
            </Heading>
            <IconAuthorList>{renderAuthors()}</IconAuthorList>
        </Wrapper>
    );
};

interface Props {}

IconAuthors.propTypes = {};
