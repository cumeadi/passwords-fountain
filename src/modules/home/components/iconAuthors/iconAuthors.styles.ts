import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { media, stylelint, theme, typography } from '@/common/theme';
setPragma(h);

export const Wrapper = styled('section')``;

export const Heading = styled('h4')`
    color: ${theme.colors.darkBlue};
    ${typography.text18}
    margin-bottom: ${theme.spacing.s12};

    ${media.gte(theme.breakpoints.s480)(stylelint.css`
        ${typography.text24}
        margin-bottom: ${theme.spacing.m18};
    `)}
`;

export const IconAuthorList = styled('ul')``;

export const IconInfo = styled('li')`
    color: ${theme.colors.darkBlue};
    ${typography.text16}

    ${media.gte(theme.breakpoints.s480)(stylelint.css`
        ${typography.text20}
    `)}
`;
