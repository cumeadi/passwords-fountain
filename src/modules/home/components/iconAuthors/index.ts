export { IconAuthors } from './iconAuthors.component';
