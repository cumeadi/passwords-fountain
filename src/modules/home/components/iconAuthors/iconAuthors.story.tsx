import { h, VNode } from 'preact';
import { IconAuthors } from './iconAuthors.component';
import { withA11y } from '@storybook/addon-a11y';

export default {
    title: 'IconAuthors',
    decorators: [withA11y],
};

export const defaultView = (): VNode => <IconAuthors />;
