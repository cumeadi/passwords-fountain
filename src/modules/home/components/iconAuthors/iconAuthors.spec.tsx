import '@testing-library/jest-dom/extend-expect';
import { h } from 'preact';
import { render } from '@testing-library/preact';
import { IconAuthors } from './iconAuthors.component';

describe('IconAuthors', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<IconAuthors />);

        expect(asFragment()).toMatchSnapshot();
    });
});
