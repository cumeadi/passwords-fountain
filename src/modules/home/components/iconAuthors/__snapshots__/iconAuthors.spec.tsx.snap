// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconAuthors should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go115679"
  >
    <h4
      class="go1694588680"
    >
      Huge thanks to icon authors:
    </h4>
    <ul
      class="go115679"
    >
      <li
        class="go2712826534"
      >
        Padlock icon - Pixel perfect
      </li>
      <li
        class="go2712826534"
      >
        Success icon - Alfredo Hernandez
      </li>
      <li
        class="go2712826534"
      >
        Error icon - Freepik
      </li>
      <li
        class="go2712826534"
      >
        Fountain icon - Smashicons
      </li>
      <li
        class="go2712826534"
      >
        Database icon - Smashicons
      </li>
      <li
        class="go2712826534"
      >
        Heart icon - Smashicons
      </li>
      <li
        class="go2712826534"
      >
        Globe icon - Freepik
      </li>
      <li
        class="go2712826534"
      >
        Recycle bin icon - Freepik
      </li>
      <li
        class="go2712826534"
      >
        Eye - Kiranshastry
      </li>
      <li
        class="go2712826534"
      >
        Github - Freepik
      </li>
      <li
        class="go2712826534"
      >
        Twitter - Pixel perfect
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
